mmengine/structures/base_data_element.py

```diff
@@ -561,6 +562,16 @@ def to_dict(self) -> dict:
             for k, v in self.all_items()
         }
 
+    def pin_memory(self) -> 'BaseDataElement':
+        """BaseDataElement supports pin_memory function to speed up
```

Comment:
need to indicate what is pin_memory in the docstring

Reply:
If we use the internal function `pin_memory`, seems this if-else is unnecessary. `pin_memory` will handle  various types of inputs.